router/throttler/adaptive.go

```diff
@@ -25,16 +25,28 @@ func (c *adaptiveThrottleConfig) readThrottlingConfig(config *config.Config, des
 		fmt.Sprintf(`Router.throttler.adaptive.%s.%s.minLimit`, destName, destID),
 		fmt.Sprintf(`Router.throttler.adaptive.%s.minLimit`, destName),
 		`Router.throttler.adaptive.minLimit`)
-	c.maxLimit = config.GetReloadableInt64Var(0, 1,
+	maxLimit := config.GetReloadableInt64Var(0, 1,
 		fmt.Sprintf(`Router.throttler.adaptive.%s.%s.maxLimit`, destName, destID),
 		fmt.Sprintf(`Router.throttler.adaptive.%s.maxLimit`, destName),
-		fmt.Sprintf(`Router.throttler.%s.%s.limit`, destName, destID),
-		fmt.Sprintf(`Router.throttler.%s.limit`, destName),
 		`Router.throttler.adaptive.maxLimit`)
+	limitMultiplier := config.GetReloadableInt64Var(150, 1,
+		fmt.Sprintf(`Router.throttler.adaptive.%s.%s.limitMultiplier`, destName, destID),
+		fmt.Sprintf(`Router.throttler.adaptive.%s.limitMultiplier`, destName),
+		`Router.throttler.adaptive.limitMultiplier`)
+	limit := config.GetReloadableInt64Var(0, 1,
+		fmt.Sprintf(`Router.throttler.%s.%s.limit`, destName, destID),
+		fmt.Sprintf(`Router.throttler.%s.limit`, destName))
+	c.maxLimit = func() int64 {
+		maxLimit := maxLimit.Load()
+		if maxLimit > 0 {
+			return maxLimit
+		}
+		return limit.Load() * limitMultiplier.Load()
```

Comment:
Either this needs to be divided by 100, or multiplier become an actual multiplier and not a percentage :)

Reply:
Fixed it